File: mural/src/main/java/com/irozon/mural/async/LoadBitmapFromURLTask.java
package com.irozon.mural.async;

import android.graphics.Bitmap;
import android.os.AsyncTask;
import android.widget.ImageView;

import com.irozon.mural.cache.ImageCache;
import com.irozon.mural.manager.ImageFetcher;

import java.lang.ref.WeakReference;
import java.net.MalformedURLException;
import java.net.URL;

/**
 * Created by Hammad Akram on 3/1/2018.
 */
public class LoadBitmapFromURLTask extends AsyncTask<String, Void, Bitmap> {
    private final WeakReference<ImageView> imageViewReference;
    private final ImageCache mCache;
    private final boolean mCacheImage;
    private final int mWidth;
    private final int mHeight;
    public String strUrl;

    public LoadBitmapFromURLTask(ImageView imageView, ImageCache cache, boolean cacheImage, int width, int height) {
        imageViewReference = new WeakReference<ImageView>(imageView);
        mCache = cache;
        mCacheImage = cacheImage;
        mWidth = width;
        mHeight = height;
    }

    @Override
    protected Bitmap doInBackground(String... urls) {
        strUrl = urls[0];
        URL url = null;
        try {
            url = new URL(strUrl);
        } catch (MalformedURLException e) {
            e.printStackTrace();
        }
        return ImageFetcher.decodeSampledBitmapFromUrl(url, mWidth, mHeight);
    }

    @Override
    protected void onPostExecute(Bitmap bitmap) {
        if (bitmap != null) {
            if (mCacheImage) {
                mCache.put(strUrl, bitmap);
            }
            final ImageView imageView = imageViewReference.get();
            if (imageView != null)
                imageView.setImageBitmap(bitmap);

        }
    }
}

File: mural/src/main/java/com/irozon/mural/async/LoadBitmapFromURLTask.kt
package com.irozon.mural.async

import android.widget.ImageView

import com.irozon.mural.cache.ImageCache
import com.irozon.mural.manager.ImageFetcher
import org.jetbrains.anko.doAsync
import org.jetbrains.anko.uiThread

import java.net.MalformedURLException
import java.net.URL

/**
 * Created by Hammad Akram on 3/1/2018.
 */
class LoadBitmapFromURLTask(private val imageView: ImageView, private val mCache: ImageCache?, private val mCacheImage: Boolean, private val mWidth: Int, private val mHeight: Int, private val strUrl: String) {

    fun execute() {
        doAsync {
            var url: URL? = null
            try {
                url = URL(strUrl)
            } catch (e: MalformedURLException) {
                e.printStackTrace()
            }
            if (url != null) {
                val bitmap = ImageFetcher.decodeSampledBitmapFromUrl(url, mWidth, mHeight)
                uiThread {
                    if (mCacheImage) {
                        mCache?.put(strUrl, bitmap)
                    }
                    imageView.setImageBitmap(bitmap)

                }
            }
        }
    }
}

